examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,99 @@
+            InitEventHubProcessor(eventHubConnection);
+            if (config is IConfigurationRoot configRoot)
+            {
+                foreach (var provider in configRoot.Providers)
+                {
+                    if (provider is Microsoft.Extensions.Configuration.AzureAppConfiguration.IConfigurationRefresher)
+                    {
+                        _refresher = (IConfigurationRefresher)provider;
+                    }
+                }
+            }
+            else
+            {
+                _logger.LogError($"{nameof(config)} is not an instance of {typeof(IConfigurationRoot)}.");
+            }
+        }
+
+        private void InitEventHubProcessor(EventHubConnection eventHubConnection)
+        {
+            if (string.IsNullOrEmpty(eventHubConnection.EventHubConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.EventHubConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.EventHubName))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.EventHubName));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.StorageConnectionString))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.StorageConnectionString));
+            }
+
+            if (string.IsNullOrEmpty(eventHubConnection.StorageContainerName))
+            {
+                throw new ArgumentNullException(nameof(eventHubConnection.StorageContainerName));
+            }
+
+            string consumerGroup = string.IsNullOrEmpty(eventHubConnection.EventHubConsumerGroup)
+                ? EventHubConsumerClient.DefaultConsumerGroupName
+                : eventHubConnection.EventHubConsumerGroup;
+
+            var storageClient = new BlobContainerClient(connectionString: eventHubConnection.StorageConnectionString,
+                                                        blobContainerName: eventHubConnection.StorageContainerName);
+
+            var processorClient = new EventProcessorClient(checkpointStore: storageClient,
+                                                       consumerGroup: consumerGroup,
+                                                       connectionString: eventHubConnection.EventHubConnectionString,
+                                                       eventHubName: eventHubConnection.EventHubName);
+
+            processorClient.ProcessEventAsync += ProcessEventHandler;
+            processorClient.ProcessErrorAsync += ProcessErrorHandler;
+            processorClient.StartProcessing();
+        }
+
+        private Task ProcessEventHandler(ProcessEventArgs eventArgs)
+        {
+            _logger.LogInformation("EventHub update received. Triggering cache invalidation.");
+
+            //
+            // Invalidate cache to force refresh.
+            _refresher?.ResetCache();
```

Comment:
Replace with the new SetDirty API

Reply:
Updated it.

---
In reply to: [462049933](https://github.com/Azure/AppConfiguration/pull/329#discussion_r462049933) [](ancestors = 462049933)